### PR TITLE
[SPECS]Stripe connect controller specs

### PR DESCRIPTION
**Pull request by apoorv-mishra**
## Notion card
NA

## Summary

Controller specs for PR-#363

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [ ] I have manually tested all workflows
- [ ] I have performed a self-review of my own code
- [ ] I have added automated tests for my code

### CODE REVIEW

File: spec/requests/payment_settings/refresh_stripe_connect_spec.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    context "when stripe connected account doesn't exist" do
+      it "is authorized to refresh stripe connect" do
+        allow(Stripe::AccountLink).to receive(:create)
+          .and_return(OpenStruct.new({ url: "https://connect.stripe.com/setup/s/something" }))
+        allow(Stripe::Account).to receive(:retrieve)
+          .and_return(OpenStruct.new({ details_submitted: false }))
+        send_request :get, payments_settings_stripe_connect_refresh_path
+        expect(response).to have_http_status(:redirect)
+        expect(response).not_to redirect_to("https://connect.stripe.com/setup/s/something")
+        expect(flash["alert"]).to eq("You are not authorized to perform this action.")
```
Comment (rohitjoshixyz) [inline spec/requests/payment_settings/refresh_stripe_connect_spec.rb:26]: Check description and assertion, it is not matching

File: spec/requests/payment_settings/refresh_stripe_connect_spec.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    context "when stripe connected account exists" do
+      it "is able to refresh stripe connect successfully" do
+        allow(Stripe::Account).to receive(:create)
+          .and_return(OpenStruct.new({ id: stripe_connected_account.account_id }))
+        allow(Stripe::AccountLink).to receive(:create)
+          .and_return(OpenStruct.new({ url: "https://connect.stripe.com/setup/s/something" }))
+        allow(Stripe::Account).to receive(:retrieve)
+          .and_return(OpenStruct.new({ details_submitted: false }))
+        stripe_connected_account.save
```
Comment (rohitjoshixyz) [inline spec/requests/payment_settings/refresh_stripe_connect_spec.rb:38]: Use bang(!) method

File: spec/requests/payment_settings/refresh_stripe_connect_spec.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+  context "when user is an admin" do
+    before do
+      create(:company_user, company:, user:)
+      user.add_role :admin, company
+      sign_in user
+    end
+
+    context "when stripe connected account doesn't exist" do
+      it "is authorized to refresh stripe connect" do
```
Comment (rohitjoshixyz) [inline spec/requests/payment_settings/refresh_stripe_connect_spec.rb:18]: Check spec description, does not refresh stripe connect account or something similar, it should match the assertion, also if the user is an admin why are they unauthorized?

**Review by rohitjoshixyz — CHANGES_REQUESTED** [review]
One spec needs changes

File: spec/requests/payment_settings/refresh_stripe_connect_spec.rb (hunk 1 above)
Comment (apoorv-mishra) [reply spec/requests/payment_settings/refresh_stripe_connect_spec.rb]: Fixed.

File: spec/requests/payment_settings/refresh_stripe_connect_spec.rb (hunk 2 above)
Comment (apoorv-mishra) [reply spec/requests/payment_settings/refresh_stripe_connect_spec.rb]: Fixed.

File: spec/requests/payment_settings/refresh_stripe_connect_spec.rb (hunk 3 above)
Comment (apoorv-mishra) [reply spec/requests/payment_settings/refresh_stripe_connect_spec.rb]: Fixed to make it return 404 now.

**Review by rohitjoshixyz — APPROVED** [review]
LGTM

**Pull request merged**